zoneinfo/src/posix.rs

```diff
@@ -136,40 +136,103 @@ pub enum PosixDate {
+                // 0 - 1.
+                let mut adjusted_week_day = week_day as i8 - days_off_week_day as i8;
+
+                // Calculate what week we are in.
+                //
+                // Since we are operating with a zero based day of month, we add
+                let week = 1 + zero_based_day_of_month / 7;
+
+                // If we have shifted beyond the month, add 7 to shift back into the first
+                // week.
+                if adjusted_week_day < 0 {
+                    adjusted_week_day += 7;
+                }
+                let week_day = WeekDay::from_u8(adjusted_week_day as u8);
+                // N.B. The left of time the target weekday becomes a time overflow added
+                // to the minutes.
+                (
+                    PosixDate::MonthWeekDay(MonthWeekDay(rule.in_month, week, week_day)),
+                    days_off_week_day as i64 * 86_400,
+                )
             }
             DayOfMonth::WeekDayLEThanMonthDay(week_day, day_of_month) => {
+                // Handle week day offset correctly
+                //
+                // We don't worry about the last day of the month in this scenario, which
+                // is the upper bound as that is handled by DayOfMonth::Last
+                let days_off_week_day = day_of_month as i8 % 7;
+                let mut adjusted_week_day = week_day as i8 - days_off_week_day;
                 let week = day_of_month / 7;
-                PosixDate::MonthWeekDay(MonthWeekDay(rule.in_month, week, week_day))
+                if adjusted_week_day < 0 {
+                    adjusted_week_day += 7;
+                }
+                (
+                    PosixDate::MonthWeekDay(MonthWeekDay(
+                        rule.in_month,
+                        week,
+                        WeekDay::from_u8(adjusted_week_day as u8),
+                    )),
+                    days_off_week_day as i64 * 86_400,
+                )
             }
         }
     }
 }
 
 #[derive(Debug, PartialEq, Clone, Copy)]
 pub struct PosixDateTime {
+    /// The designated [`PosixDate`]
     pub date: PosixDate,
+    /// The local time for a [`PosixDateTime`] at which a transition occurs.
+    ///
+    /// N.B., this can be in the range of -167..=167
     pub time: Time,
 }
 
 impl PosixDateTime {
     pub(crate) fn from_rule_and_transition_info(rule: &Rule, offset: Time, savings: Time) -> Self {
-        let date = PosixDate::from_rule(rule);
+        let (date, time_overflow) = PosixDate::from_rule(rule);
         let time = match rule.at {
-            QualifiedTime::Local(time) => time,
-            QualifiedTime::Standard(standard_time) => standard_time.add(rule.save),
-            QualifiedTime::Universal(universal_time) => universal_time.add(offset).add(savings),
+            QualifiedTime::Local(time) => time.add(Time::from_seconds(time_overflow)),
```

Comment:
nit: probably just add to the time outside the `match`, `let time = time.add(overflow)`